qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -206,7 +200,80 @@ def run(self, dag):
             self.property_set["layout"] = initial_layout
             self.routing_pass.fake_run = False
             return dag
-        dist_matrix = self.coupling_map.distance_matrix
+        # Combnined
+        layout_components = disjoint_utils.run_pass_over_connected_components(
+            dag, self.coupling_map, self._inner_run
+        )
+        initial_layout_dict = {}
+        final_layout_dict = {}
+        for (
+            layout_dict,
+            final_dict,
+            component_map,
+            _gate_order,
+            _swap_map,
+            _dag,
+        ) in layout_components:
+            initial_layout_dict |= {k: component_map[v] for k, v in layout_dict.items()}
+            final_layout_dict |= {component_map[k]: component_map[v] for k, v in final_dict}
```

Comment:
Sigh this syntax is only on python>=3.9 which is causing the CI failure. I'll have to use `.update()`  here instead.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9802/commits/f069f669e503e371101f9792e1c717d285885b2c